### PR TITLE
Finishing the Source class

### LINKED ISSUE #44: Finish Context and Source
Try to do as many TODO as you are able to do, gradually.

### PR DESCRIPTION
**Pull request by Huy-Ngo**
I haven't tested this yet, but I guess that is the problem of #46  now.

In this PR, I:

- Make a `make_filter_params` to convert from `Tuple[float, float, float]` to `alure::FilterParams`.
- Finish the 3 TODOs in `Source`: `direct_filter`, `send_filter`, and `auxiliary_send_filter`.

This PR will resolve #44.

### CODE REVIEW

File: src/util.h [hunk 1]
```diff
@@ -124,6 +124,16 @@ namespace palace
...
   }
 
+  inline alure::FilterParams
+  make_filter_params (std::vector<float> params) noexcept
+  {
+    alure::FilterParams filter_params;
+    filter_params.mGain = params[0];
+    filter_params.mGainHF = params[1];
+    filter_params.mGainLF = params[2];
+    return filter_params;
```
Comment (McSinyx) [inline src/util.h]: Is it possible to `return alure::FilterParams {params[0], params[1], params[2]};`?

**Review by McSinyx — COMMENTED** [review]
LGTM, aside a minor comment.

**Comment by McSinyx** [comment]
Hugging Travis don't even have macOS available ugh.  BTW don't forget to add your name to the copyright headers of files you modify.

File: src/util.h [hunk 2]
```diff
@@ -124,6 +125,12 @@ namespace palace
     return attributes;
   }
 
+  inline alure::FilterParams
+  make_filter_params (std::vector<float> params) noexcept
+  {
+    return alure::FilterParams {params[0], params[1], params[2]};
+  }
```
Comment (McSinyx) [inline src/util.h:132]: You might want to join the three lines together, otherwise it's OK to squash and merge now.

**Review by McSinyx — APPROVED** [review]

**Comment by McSinyx** [comment]
It's time to throw a release party :tada:  I'll handle Reddit, you can do Discord, @9a24f0 can do any social media perse can think of.

**Comment by Huy-Ngo** [comment]
Hmm idk, how about Twitter and LinkedIn? I have my accounts, but I'm relatively inactive there. Also have instagram, but I don't think it's a fitting social media.

Anyway yay 🎉

**Comment by McSinyx** [comment]
If you think it fits d-:

**Pull request merged**